Code/Framework/AzCore/AzCore/variadic.h

```diff
@@ -40,7 +40,7 @@
 
 #   define AZ_VA_NUM_ARGS(...) AZ_VA_NUM_ARGS_SUB(__VA_ARGS__)
 
-#if AZ_HAS_VA_OPT
+ #if AZ_HAS_VA_OPT && !FORCE_VA_OPT_ARG
```

Comment:
Why was this changed?

Reply:
This is required so that shiboken can build the wrapper files correctly, if not done, clang gives the error
`1>C:/ly/git/o3de/code/Framework/AzCore\AzCore/variadic.h(57,1): error G1A2B8E67: static_assert failed due to requirement '1 == 2'`

Reply:
Which version of clang is this, because we use the same preprocessor of clang on Mac, iOS, Linux and Android without any issues?

Reply:
It's all the versions I've tried: on further investigation it looks like it's happening because of the -fno-compatibility argument that is being automatically inserted by shiboken-generator. If I remove that from the command line the error disappears. I'll see if I can find a way to force shiboken to stop doing it and whether this breaks anything else.

Reply:
As the argument is added automatically by shiboken, it's impossible to remove without patching it. The clang-options command line argument doesn't get added until version 6. I'll look into how practical patching is.

Reply:
Ok, patched exe works. Reverting this change.